src/v/kafka/server/replicated_partition.cc

```diff
@@ -215,7 +215,8 @@ replicated_partition::aborted_transactions(
     }
     if (
       _partition->cloud_data_available()
-      && offsets.begin_rp < _partition->start_offset()) {
+      && offsets.begin
+           < _translator->from_log_offset(_partition->start_offset())) {
         // The fetch request was satisfied using shadow indexing.
```

Comment:
Great find! Maybe it's worth to extract a function that can be used to check what part of the partition should we read from (remote or local) so we won't hit this bug in the future when new code will need to make the same decision.

Reply:
:+1: I've gone ahead and added a commit that does this.